PyRDF/Proxy.py

```diff
@@ -64,14 +106,62 @@ def GetValue(self):
             current action node in the computational graph.
 
         """
-        if not self.action_node.value:  # If event-loop not triggered
+        if not self.proxied_node.value:  # If event-loop not triggered
             from . import current_backend
-            generator = CallableGenerator(self.action_node.get_head())
+            generator = CallableGenerator(self.proxied_node.get_head())
             current_backend.execute(generator)
 
-        return self.action_node.value
+        return self.proxied_node.value
 
     def _call_handler(self, *args, **kwargs):
         # Handles an operation call to the current action node
         # and returns result of the current action node.
         return getattr(self.GetValue(), self._cur_attr)(*args, **kwargs)
+
+
+class TransformationProxy(Proxy):
+    """
+    A proxy object to an instantiated node. Used as a controller of the user
+    references to the node itself. When the user deletes reference to a
+    node (e.g. assigning the same variable to another operation), the proxy
+    object will get destroyed by Python, thus flagging the node to be without
+    user references anymore.
+    """
+
+    def __del__(self):
+        """Deletes current Proxy and flags the wrapped Node for pruning"""
+        self.proxied_node.has_user_references = False
+
+    def __getattr__(self, attr):
+        """
+        Intercepts calls on operation or attributes belonging to the proxied
+        node.
+
+        Returns either:
```

Comment:
This should be documented like [here](https://github.com/JavierCVilla/PyRDF/blob/master/PyRDF/RDataFrame.py#L127-L134).